lib/dialer/internal.go

```diff
@@ -78,8 +78,26 @@ func dialWithFallback(proxyDialFunc dialFunc, fallbackDialFunc dialFunc, network
 	return conn, err
 }
 
+/// This is a rip off of proxy.FromURL for "socks" URL scheme
+func socksDialerFunction(u *url.URL, forward proxy.Dialer) (proxy.Dialer, error) {
+	var auth *proxy.Auth
+	if u.User != nil {
+		auth = new(proxy.Auth)
+		auth.User = u.User.Username()
+		if p, ok := u.User.Password(); ok {
+			auth.Password = p
+		}
+	}
+
+	return proxy.SOCKS5("tcp", u.Host, auth, forward)
+}
+
 // This is a rip off of proxy.FromEnvironment with a custom forward dialer
-func getDialer(forward proxy.Dialer) proxy.Dialer {
+func getDialer(forward proxy.Dialer, initCall bool) proxy.Dialer {
+	if initCall {
+		proxy.RegisterDialerType("socks", socksDialerFunction)
```

Comment:
You can move this to an [`init` func](https://medium.com/golangspec/init-functions-in-go-eac191b3860a)

Reply:
I thought about that but doesn't functions used in variable initialization get called before init()? As we get usingProxy variable by calling getDialer in variable initialization, I'd have to essentially copy that code into init() after calling RegisterDialerType. Something like that? https://pastebin.com/39qWkWuL
I may be wrong of course, I'm not a golang expert.

Reply:
Or I guess I could just move the proxyDialer and usingProxy initialization from var ( ) block to init()